be/src/io/fs/s3_file_system.cpp

```diff
@@ -226,7 +229,26 @@ Status S3FileSystem::delete_directory(const Path& path) {
 }
 
 Status S3FileSystem::link_file(const Path& src, const Path& dest) {
-    return Status::NotSupported("not support");
+    auto client = get_client();
+    CHECK_S3_CLIENT(client);
+
+    Aws::S3::Model::CopyObjectRequest request;
+    auto src_key = get_key(src);
+    auto dest_key = get_key(dest);
+    request.WithBucket(_s3_conf.bucket)
+            .WithCopySource(_s3_conf.bucket + "/" + src_key)
```

Comment:
Pls do not use copy to implement link, it is very expensive and not assure atomic and also there will be garbage if it failed. 
Should read and write a new rowset at application layer and then we could do gc.